odl/solvers/advanced/proximal_operators.py

```diff
@@ -374,11 +738,60 @@ def _call(self, x, out):
                 # Pointwise division
                 out.divide(diff, out)
 
-    return _ProximalConvConjL1
+    return ProximalConvConjL1
+
+
+def proximal_l1(space, lam=1, g=None, isotropic=False):
+    """Proximal operator factory of the l1-norm.
+
+    Function for the proximal operator of the functional F where F is an
+    l1-norm
+
+        F(x) = lam ||x - g||_1
+
+    with x and g elements in ``space``, and scaling factor lam.
+
+    The proximal operator of ``tau * F`` where ``tau`` is a step size is
+
+                              y - tau * lam   if y > tau * lam,
+         prox[tau * F](y) = { 0               if -tau * lam <= y <= tau * lam
+                              y + tau * lam   if y < -tau * lam
+
+    An alternative formulation is available for `ProductSpace`'s, in that case
+    the ``isotropic`` parameter can be used, giving
+
+        F(x) = lam || ||x - g||_2 ||_1
+
+    Where the proximal can be calculated using the Moreau equality.
```

Comment:
Im lazy here, couldnt get it into a simple form. I guess this is not hugely important.